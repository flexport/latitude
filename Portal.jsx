/**
 * TEAM: frontend_infra
 * @flow strict
 */
import * as React from "react";
import ReactDOM from "react-dom";
import invariant from "./tools/invariant";

type Props = {|
  +children: React.Node,
|};

/**
 * @short Wrap Popper components with Portal for popper overlay management.
 * @category Layout
 * @status Beta
 * @brandStatus V3
 *
 * Wrapper to place any element into the body of the dom. Should be used for
 * components like a Takeover.
 */
function Portal({children}: Props): React.Portal {
  invariant(document.body !== null);
  return ReactDOM.createPortal(children, document.body);
}

export default Portal;
