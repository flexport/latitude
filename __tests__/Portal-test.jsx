/**
 * TEAM: frontend_infra
 *
 * @flow
 */

import * as React from "react";
import {mount} from "enzyme";
import Portal from "../Portal";

function mountPortal(children: React.Node) {
  return mount(
    <div>
      <Portal>{children}</Portal>
    </div>
  );
}

describe("Portal", () => {
  it("matches snapshot", () => {
    const comp = mountPortal(null);
    expect(comp).toMatchSnapshot();
  });

  it("uses portals", () => {
    const comp = mountPortal(null);

    expect(comp.find("Portal").exists()).toBe(true);
  });
});
