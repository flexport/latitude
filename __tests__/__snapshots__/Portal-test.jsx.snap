// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal matches snapshot 1`] = `
<div>
  <Portal>
    <Portal
      containerInfo={<body />}
    />
  </Portal>
</div>
`;
